Settings based docs

mihai-vlc opened the pull request:
These changes will:
- Add menu options for the key bindings and settings files in Preferences -> Package Settings -> GotoDocumentation
- Has a dictionary defined at the top of the file with the default docs
- Merge the default docs with the ones defined in the settings file
- Adds the `%(query)s` and `%(scope)s` placeholders
- Adds the option to default to google in case the scope is not defined
- If a scope has a the value false it will consider the scope disabled and it will not perform a search

Tested on:
- ST3 - Win7 x64
- ST2 - Win XP x32

mihai-vlc committed:
Implements the new settings logic, adds the default settings, adds options in the menu

kemayo commented on goto_documentation.py:
```diff
@@ -0,0 +1,182 @@
+}
+
+
+def combineDicts(dictionary1, dictionary2):
```
Why not just do:

```
new_dict = dictionary1.copy()
new_dict.update(dictionary2)
return new_dict
```

We don't need a deep copy here (it seems reasonable to say someone overriding part of the `python` definition might as well override the whole thing), and that's far easier to understand.

kemayo commented on goto_documentation.py:
```diff
@@ -0,0 +1,182 @@
+        "scss": "http://devdocs.io/#q=%(scope)s+%(query)s",
+        "less": "http://devdocs.io/#q=%(scope)s+%(query)s",
+        "python": {
+            "command": "python -m pydoc %(query)s",
```
This should probably be: `"command": ["python", "-m", "pydoc", "%(query)s",`

...and the code below should account for that with applying the substitution. I think this being a single string is why I'm getting errors looking for python documentation with this.

kemayo commented on goto_documentation.sublime-settings:
```diff
@@ -0,0 +1,50 @@
+        // supported placeholders:
+        //  - %(query)s the selected text/word
+        //  - %(scope)s the current scope
+        "ahk": "http://www.autohotkey.com/docs/commands/%(query)s.htm",
```
These are all no longer necessary, with the `default_docs` dict being present.

mihai-vlc replied:
I think we should keep them there as a reference, not everybody will open the *.py file.
And it's easier to copy/paste an item from there if you want to extend the docs.

kemayo replied:
We can just leave a single one there commented out, and that'll serve the same purpose.

kemayo commented on goto_documentation.py:
```diff
@@ -0,0 +1,182 @@
+                else:
+                    self.show_status("This scope is disabled !")
+
+        else:
```
It occurs to me that this could be rewritten as a doc scope of its own. Then instead of a special branch here we just do:

```
if not scope in docs:
    scope = settings.get('fallback_scope', 'google')
if not scope in docs:
    return

# continue as written from here...
```

That'd unify some things, and make it so we need less special-cased code.

mihai-vlc committed:
Changes the logic for the fallback url, adds the command into a list

mihai-vlc committed:
Updated the readme file with the new settings

mihai-vlc commented:
I have updated the code
- Changed the logic to use the fallback scope
- Removed the docs from the settings file (kept only css and python as examples)
- Replaced the recursive merge with the one level one
